independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/interceptors/aroundconstruct/AroundConstructAppliedViaConstructorTest.java

```diff
@@ -0,0 +1,86 @@
+package io.quarkus.arc.test.interceptors.aroundconstruct;
+
+import static org.junit.Assert.assertNotNull;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.test.ArcTestContainer;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.interceptor.AroundConstruct;
+import javax.interceptor.Interceptor;
+import javax.interceptor.InvocationContext;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class AroundConstructAppliedViaConstructorTest {
```

Comment:
Might be worth expending this so that there are two interceptors applied simultaneously - one with class level and one with constructor level binding.

Reply:
Done